base/iterators.jl

```diff
@@ -1077,6 +1077,7 @@ else
     # fixpoint.
     function fixpoint_iter_type(itrT::Type, valT::Type, stateT::Type)
         nextvalstate = Base._return_type(next, Tuple{itrT, stateT})
+        nextvalstate === Union{} && return Any
```

Comment:
Cc @Keno — I coupled this change here.  The failure mode is `join((), ",")`, which attempts to find the value type of `next((), start(()))` — that's an error and will always be `Union{}`.  I'm assuming that a case like this was intended to be caught by the next line, but of course the bottom type is a subtype of everything so it flies on by and throws an error later.  Would it be better to return `Any` or `Union{}` in this case?

Reply:
My stab in the dark here didn't work.  I've just marked things as `@test_broken` and opened #26871.